src/rsvc/srv_common.c

```diff
@@ -255,6 +262,18 @@ ds_rsvc_get_attr(struct ds_rsvc *svc, struct rdb_tx *tx, rdb_path_t *path,
 		} else {
 			iovs[j].iov_buf_len = sizes[i];
 			sizes[i] = iovs[j].iov_len;
+
+			/* need to copy from pmem to ram to avoid errors
+			 * during RDMA registration
+			 */
+			D_ALLOC(new_buf, iovs[j].iov_len);
```

Comment:
Any reason not to allocate a single buffer for all of them?  e.g. calculate the size first and then do a single allocation and copy bits to it.   I think bulk registration would be more efficient that way.

Reply:
> I think bulk registration would be more efficient that way.

Well, it could only be if the remote buffers are also contiguous, right ?
But taking a closer look to it, an other way to improve this piece of code would be to do the D_ALLOC() an memcpy() only if there is a remote buffer and to do th memcpy() with the min(sgl.sg_iovs[j].iov_len,  sgl.sg_iovs[j].iov_buf_len) as the size.